lib/hd/keyorigin.js

```diff
@@ -0,0 +1,173 @@
+/*!
+ * keyorigin.js - hd key path object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('assert');
+const common = require('./common');
+const bio = require('bufio');
+const {inspectSymbol} = require('../utils');
+/**
+ * Helper class to represent hd key path for arbitrary wallets.
+ * @property {Number} fingerPrint - master key fingerprint (uint32)
+ * @property {Array} path - bip32 derivation path in uint32 array
+ */
+class KeyOriginInfo {
+  constructor(options) {
+    this.fingerPrint = -1;
```

Comment:
I don't think this is a great default value, example:

```
> new bcoin.hd.hd.KeyOriginInfo()
{ fingerPrint: '000000-1', path: 'm' }
```

In `private.js` for example we initialize:

https://github.com/bcoin-org/bcoin/blob/b0058696cc10c8f9b17190b31fd2cd907d85d047/lib/hd/private.js#L54

Reply:
Agreed, good catch